src/App.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import RestaurantList from './components/RestaurantList';
+import mockData from "./data/mockData.json";
+import { Category, Restaurant, Sort } from './types/Restaurant';
+import SelectBox from './components/SelectBox';
+import Header from './components/Header';
+import styled from 'styled-components';
+
+interface AppState {
+  restaurants: Restaurant[];
+  sortBy: Sort;
+  categorizeBy: Category;
+}
+class App extends React.Component<{}, AppState> {
+
+  constructor(props: Readonly<{}> | {}) {
+    super(props);
+    this.state = { restaurants: mockData, sortBy: 'name', categorizeBy: 'all' };
+  }
```

Comment:
선언된 클래스 컴포넌트의 메서드는 `this`의 바인딩이 정상적으로 이루어지지 않을 수 있으니, 생성자 함수에서 바인딩해두는 것을 추천드립니다.

```suggestion
  constructor(props: Readonly<{}> | {}) {
    super(props);
    this.state = { restaurants: mockData, sortBy: 'name', categorizeBy: 'all' };
    this.updateSortBy = this.updateSortBy.bind(this);
    this.updateCategorizeBy = this.updateCategorizeBy.bind(this);
  }
```